DigitalLearningSolutions.Web.Tests/Controllers/LearningMenu/LearningMenuControllerTests.cs

```diff
@@ -218,13 +224,94 @@ public void Tutorials_should_StartOrUpdate_course_sessions()
                 .MustNotHaveHappened();
         }
 
+        [Test]
+        public void Tutorials_should_not_StartOrUpdate_course_sessions_if_invalid_tutorial()
+        {
+            // Given
+            A.CallTo(() => tutorialContentService.GetTutorialContent(CandidateId, CustomisationId, SectionId, TutorialId))
+                .Returns(null);
+            A.CallTo(() => courseContentService.GetOrCreateProgressId(CandidateId, CustomisationId, CentreId)).Returns(1);
+
+            // When
+            controller.Tutorial(CustomisationId, SectionId, TutorialId);
+
+            // Then
+            A.CallTo(() => sessionService.StartOrUpdateSession(A<int>._, A<int>._, A<ISession>._)).MustNotHaveHappened();
+        }
+
+        [Test]
+        public void Tutorials_should_not_StartOrUpdate_course_sessions_if_progress_missing()

```

Comment:
Minor but I think "unable to create progress" or "unable to enrol" is more descriptive. Because if the progress is just missing then we should enrol them on the course 🙂

Reply:
Yeah I agree. Renamed in 5c1c564 🙂